test/map_notifier_test.dart

```diff
@@ -131,41 +163,60 @@ void main() {
 
       test('Listener is not notified if no value is removed', () {
         mapNotifier.addAll(newValues);
+
+        /// first call when addAll is called
         expect(listenerCallCount, 1);
         mapNotifier.remove('zero');
+
+        /// second call when remove is called
         expect(listenerCallCount, 2);
         mapNotifier.remove('zero');
+
+        /// The count is still 2 because the value is already removed
         expect(listenerCallCount, 2);
       });
 
       test(
         'Listener is not notified if removeWhere does not remove any values',
         () {
           mapNotifier.addAll(newValues);
+
+          /// first call when addAll is called
           expect(listenerCallCount, 1);
           mapNotifier.removeWhere((key, _) => key == 'ten');
+
+          /// The count is still 1 because the value is not excited in mapNotifier
           expect(listenerCallCount, 1);
         },
       );
 
       test(
           'Listener is not notified when update is called and updates to the '
           'already existing value', () {
+        /// The key exist in mapNotifier, so the value is not added
         mapNotifier.update('zero', (_) => 10, ifAbsent: () => 10);
-        expect(listenerCallCount, 1);
+
+        /// The listener isn't called because the value is already updated
+        expect(listenerCallCount, 0);
```

Comment:
Nope, what you are doing to fix that test is changing the test and not checking why it is failing. Because if you read the description on this function and check what the initial condition of this test is (the setup method is called before every test and the tearDown function after it) then this would update the key's value to 10 and therefore should call the listener.

Reply:
Also as the new default behaviour of MapNotifier is to update on any update even if the value doesn't change because of 
```dart
    CustomNotifierMode notificationMode = CustomNotifierMode.always,
```
in the constructor definition of MapNotifier

to test this behaviour correctly you have to create a new MapNotifier at the beginning of the test with `notificationMode= normal`